pyomo/solvers/tests/checks/test_GAMS.py

```diff
@@ -162,5 +165,146 @@ def test_keepfiles_gms(self):
 
             shutil.rmtree(tmpdir)
 
+class GAMSLogfileTestBase(unittest.TestCase):
+    def setUp(self):
+        """Set up model and temporary directory."""
+        m = ConcreteModel()
+        m.x = Var()
+        m.c = Constraint(expr= m.x >= 10)
+        m.o = Objective(expr= m.x)
+        self.m = m
+        self.tmpdir = mkdtemp()
+        self.logfile = os.path.join(self.tmpdir, 'logfile.log')
+
+    def tearDown(self):
+        """Clean up temporary directory after tests are over."""
+        shutil.rmtree(self.tmpdir)
+
+    def _check_logfile(self, exists=True, expected=None):
+        """Check for logfiles existence and contents.
+
+        exists=True:
+            Whether to check if the logfile exists or doesn't exist.
+        expected=None:
+            Optionally check that the logfiles contents is equal to this value.
+
+        """
+        if not exists:
+            self.assertFalse(os.path.exists(self.logfile))
+            return
+
+        self.assertTrue(os.path.exists(self.logfile))
+        with open(self.logfile) as f:
+            logfile_contents = f.read()
+        self.assertTrue(logfile_contents)
+        if expected:
+            self.assertEqual(logfile_contents, expected)
+
+
+@unittest.skipIf(not gamsgms_available, "The 'gams' executable is not available")
+class GAMSLogfileGmsTests(GAMSLogfileTestBase):
+    """Test class for testing permultations of tee and logfile options.
+
+    The tests build a simple model and solve it using the different options
+    using the gams command directly.
+
+    """
+
+    def test_no_tee(self):
+        with SolverFactory("gams", solver_io="gms") as opt:
+            with redirected_subprocess_call() as output:
+                opt.solve(self.m, tee=False)
+        self.assertFalse(output.getvalue())
+        self._check_logfile(exists=False)
+
+    def test_tee(self):
+        with SolverFactory("gams", solver_io="gms") as opt:
+            with redirected_subprocess_call() as output:
+                opt.solve(self.m, tee=True)
+        self.assertTrue(output.getvalue())
+        self._check_logfile(exists=False)
+
+    def test_logfile(self):
+        with SolverFactory("gams", solver_io="gms") as opt:
+            with redirected_subprocess_call() as output:
+                opt.solve(self.m, logfile=self.logfile)
+        self.assertFalse(output.getvalue())
+        self._check_logfile(exists=True)
+
+    def test_tee_and_logfile(self):
+        with SolverFactory("gams", solver_io="gms") as opt:
+            with redirected_subprocess_call() as output:
+                opt.solve(self.m, logfile=self.logfile, tee=True)
+        self.assertTrue(output.getvalue())
+        self._check_logfile(exists=True, expected=output.getvalue())
```

Comment:
Given the suggested change from `subprocess.call` to `pyutilib.subprocess.run` in `GAMSShell.solve()`, we can convert all these from `redirected_subprocess_call` to `redirected_stdout`

Reply:
converted `subprocess.call` to `pyutilib.subprocess.run` but left the test calls as they were (see comment below)